pkg/oc/bootstrap/clusterup/components/router/router_install.go

```diff
@@ -0,0 +1,186 @@
+		return nil
+	}
+	if !apierrors.IsNotFound(err) {
+		return errors.NewError("error retrieving router service").WithCause(err)
+	}
+
+	componentName := "install-router"
+	imageRunHelper := run.NewRunHelper(dockerhelper.NewHelper(dockerClient)).New()
+	glog.Infof("Running %q", componentName)
+
+	// Create service account for router
+	routerSA := &kapi.ServiceAccount{}
+	routerSA.Name = RouterServiceAccountName
+	_, err = kubeClient.Core().ServiceAccounts("default").Create(routerSA)
+	if err != nil {
+		return errors.NewError("cannot create router service account").WithCause(err)
+	}
+
+	// Add router SA to privileged SCC
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		securityClient, err := securityclientinternal.NewForConfig(restConfig)
+		if err != nil {
+			return err
+		}
+		privilegedSCC, err := securityClient.Security().SecurityContextConstraints().Get("privileged", metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		privilegedSCC.Users = append(privilegedSCC.Users, serviceaccount.MakeUsername("default", RouterServiceAccountName))
+		_, err = securityClient.Security().SecurityContextConstraints().Update(privilegedSCC)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		return errors.NewError("cannot update privileged SCC").WithCause(err)
+	}
+
+	// Create router cert
+	cmdOutput := &bytes.Buffer{}
+	createCertOptions := &admin.CreateServerCertOptions{
+		SignerCertOptions: &admin.SignerCertOptions{
+			CertFile:   filepath.Join(c.MasterConfigDir, "ca.crt"),
+			KeyFile:    filepath.Join(c.MasterConfigDir, "ca.key"),
+			SerialFile: filepath.Join(c.MasterConfigDir, "ca.serial.txt"),
+		},
+		Overwrite: true,
+		Hostnames: []string{
+			c.RoutingSuffix,
+			// This will ensure that routes using edge termination and the default
+			// certs will use certs valid for their arbitrary subdomain names.
+			fmt.Sprintf("*.%s", c.RoutingSuffix),
+		},
+		CertFile: filepath.Join(c.MasterConfigDir, "router.crt"),
+		KeyFile:  filepath.Join(c.MasterConfigDir, "router.key"),
+		Output:   cmdOutput,
+	}
+	_, err = createCertOptions.CreateServerCert()
+	if err != nil {
+		return errors.NewError("cannot create router cert").WithCause(err)
+	}
+
+	err = catFiles(filepath.Join(c.MasterConfigDir, "router.pem"),
+		filepath.Join(c.MasterConfigDir, "router.crt"),
+		filepath.Join(c.MasterConfigDir, "router.key"),
+		filepath.Join(c.MasterConfigDir, "ca.crt"))
+	if err != nil {
+		return errors.NewError("cannot create aggregate router cert").WithCause(err)
+	}
+
+	routerCertPath := c.MasterConfigDir + "/router.pem"
+	flags := []string{
+		"adm", "router",
+		"--host-ports=true",
+		"--loglevel=8",
+		"--config=" + c.MasterConfigDir + "/admin.kubeconfig",
+		fmt.Sprintf("--host-network=%v", !c.PortForwarding),
+		fmt.Sprintf("--images=%s", c.ImageFormat),
+		fmt.Sprintf("--default-cert=%s", routerCertPath),
+	}
+	_, stdout, stderr, rc, err := imageRunHelper.Image(c.OCImage).
+		Privileged().
```

Comment:
This is just running oc create right? Does it have to be privileged?

Reply:
> This is just running oc create right? Does it have to be privileged?

Yeah, it fails otherwise.  It can't read some files.